ui/src/app/views/application/show/workflow/application.workflow.component.ts

```diff
@@ -193,29 +191,39 @@ export class ApplicationWorkflowComponent implements OnInit {
      * Action when changing branch
      */
     changeBranch(): void {
-        // reinit verison filter
-        this.applicationFilter.version = '';
-        jQuery('.cdsVersion div.text')[0].textContent = '';
-        this.changeVersion();
-
         // Load the versions of the new branch
-        this.loadVersions(this.project.key, this.application.name);
+        this.loadVersions(this.project.key, this.application.name)
+            .subscribe(() => this.changeVersion());
     };
 
     /**
      * Action when changing version
      */
-    changeVersion(): void {
+    changeVersion(version?: number): void {
         this.applicationFilter.branch = this.applicationFilter.branch.trim();
-        if (this.applicationFilter.version.trim() === '') {
-            this.applicationFilter.version = 0;
+
+        if (!version && Array.isArray(this.versions) && this.versions.length) {
+            this.applicationFilter.version = this.versions[1];
+        }
+
+        if (version) {
+            this.applicationFilter.version = version;
         }
+
         this._router.navigate(['/project/', this.project.key, 'application', this.application.name],
             {queryParams: {tab: 'workflow', branch: this.applicationFilter.branch, version: this.applicationFilter.version}});
         this.changeWorkerEvent.emit(true);
         this.clearTree(this.application.workflows);
     }
 
+    /**
+     * Load the list of version for the current application on the selected branch
+     */
+    loadVersions(key: string, appName: string): Observable<Array<number>> {
+        return this._appWorkflow.getVersions(key, appName, this.applicationFilter.branch)
+            .map((versions) => this.versions = versions);
+    };
+
```

Comment:
The default version filter isn't the last version ? I mean, when your version filter is empty the version displayed is the last version ? Here the behavior is that when you don't select any filter it's empty. I don't understand why it is a problem to display the last version when you select a branch because it's the same behavior than before except that we have more information displayed and that give us the ability to kick jquery. I think there is a detail I don't catch. If you want I'm available on Gitter